schema/chat_room.proto

```diff
@@ -0,0 +1,343 @@
+      patch: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      body: "*"
+      additional_bindings: {
+        patch: "/v1alpha3/{parent=global}/{name=messages/*}"
+      }
+    };
+  }
+
+  rpc DeleteMessage(DeleteMessageRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      additional_bindings: {
+        patch: "/v1alpha3/{parent=global}/{name=messages/*}"
+      }
+    };
+  }
+
+  rpc ListMessages(ListMessagesRequest) returns (ListMessagesResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{parent=rooms/*}/messages"
+      additional_bindings: {
+        get: "/v1alpha3/{parent=global}/messages"
+      }
+    };
+  }
+
+  rpc StreamMessages(stream CreateMessageRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{parent=rooms/*}:streamMessages"
+      body: "*"
+      additional_bindings: {
+        post: "/v1alpha3/{parent=global}:streamMessages",
+        body: "*"
+      }
+    };
+  }
+
+  rpc Chat(stream CreateMessageRequest) returns (stream Message);
+}
+
+message Room {
+  string name = 1 [
+    (google.api.resource_set) = {
+      base_name: "Room"
+      resources: [
+        {base_name: "UserRoom"    path: "rooms/*"},
+        {base_name: "GlobalRoom"  path: "global"}
+      ]
+    }];
+
+  string display_name = 2;
+
+  string description = 3;
+
+  google.protobuf.Timestamp create_time = 4;
+
+  google.protobuf.Timestamp update_time = 5;
+}
+
+message CreateRoomRequest {
+  Room room = 1;
+}
+
+message GetRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+}
+
+message UpdateRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+
+  Room room = 2;
+}
+
+message DeleteRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+}
+
+message ListRoomsRequest {
+  int32 page_size = 1;
+
+  string page_token = 2;
+}
+
+message ListRoomsResponse {
+  repeated Room rooms = 1;
+
+  string next_page_token = 2;
+}
+
+message StreamRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
+  ];
+}
+
+message SearchAllRequest {
+  string query = 1 [(google.api.required) = true];
+}
+
+message SearchAllMetadata {
+  google.rpc.RetryInfo retry_info = 1;
+}
+
+message SearchAllResponse {
+  repeated Message messages = 1;
+}
+
+message Message {
+  string parent = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room"];
+
+  string name = 2 [(google.api.resource).path = "messages/*"];
+
+  // This will eventually become a User message that will cover add coverage for
+  // shared resources.
+  string author = 3;
+
+  string content = 4;
```

Comment:
Perhaps wrap `content` and `image` in a oneof? I have not seen any oneofs yet.

Reply:
Done!